File: SUM5/SUM5.py
T = int(input())

queries = []
for _ in range(T):
    queries.append(int(input()))

max_n = max(queries)

H = [0.0] * (max_n + 1)
for i in range(1, max_n + 1):
    H[i] = H[i-1] + 1.0 / i

for n in queries:
    print("{:.5f}".format(H[n]))

File: SUM5/SUM5.cpp
#include<bits/stdc++.h>
using namespace std;

int main() {
    int T;
    cin >> T;
    vector<int> queries;
    for (int _ = 0; _ < T; _++) {
        int n;
        cin >> n;
        queries.push_back(n);
    }
    int max_n = *max_element(queries.begin(), queries.end());
    vector<double> H(max_n + 1, 0.0);
    for (int i = 1; i <= max_n; i++) {
        H[i] = H[i-1] + 1.0 / i;
    }
    for (int n : queries) {
        cout << fixed << setprecision(5) << H[n] << endl;
    }
    return 0;
}
